src/type-system.ts

```diff
@@ -354,6 +361,33 @@ export const ElysiaType = {
 			})
 			.Encode((value) => value) as any as TNumber
 	},
+	Integer: (property?: NumberOptions): TInteger => {
+		const schema = Type.Integer(property)
+
+		return t
+			.Transform(
+				t.Union(
+					[
+						t.String({
+							format: 'integer',
+							default: 0
+						}),
+						t.Number(property)
+					],
+					property
+				)
+			)
+			.Decode((value) => {
+				const number = +value
+				if (isNaN(number)) return value
```

Comment:
`nan` is not a valid integer usually

Reply:
applied in 3173e52281ce11025b3567d0a74ecf0298636a2c